docs/components/aws/elasticache.md

```diff
@@ -0,0 +1,792 @@
+instances can be selected from Existing cluster dropdown.
+
+<p align="center">
+  <img style="Padding:20px" width="400" src="./images/elasticache/ecache_existing_cluster.png">
+</p>
+
+### Basic Code Template
+
+After ElastiCache configuration completed, Sigma IDE will generate following basic code template for you. 
+
+```
+// You must always quit the redis client after it's used
+redis.[operation-name]({
+    clusterIdentifier: 'sigma-redis',
+    params: [
+        //Operation parameters
+    ]
+}, function (error, response, redisClient) {
+    if (error) {
+        callback(error);
+    } else {
+        //redisClient.quit();
+    }
+});
+```
+
+### Perform operations on multiple keys using single Sigma operation
+
+Most of the Redis operations are restricted to interact with only one key inside a single operation. 
+With Sigma IDE, now you can perform same redis operation on multiple of keys using single Sigma operation. 
+
+For an example redis `get` operation returns value of corresponding key while Sigma allows
+you to fetch values of multiple keys using single Sigma get operation.
+
+<p align="center">
+  <img style="Padding:20px" width="400" src="./images/elasticache/ecache_multikey_operation.png">
+</p>
+
+```
+redis.get({
+    clusterIdentifier: 'sigma-redis',
+    params: [
+        'key1', 'key2', 
+    ]
+}, function (error, response, redisClient) {
+    if (error) {
+        callback(error);
+    } else {
+        //redisClient.quit();
+    }
+});
+```
+
+### Return Values 
+
+For every executed operation Sigma will return following attributes in its callback function.
+
+ - `error`: This object will contain error if any. Otherwise it will be `null`
+ - `response`: Response object of the operation. 
+ - `redis client`: Redis client connection object that created communicate with the Redis server.
+
+### Response Format
+
+Field   | Type   | Description  
+:---:   | :---:  | :---
+Success | Number | Number of succeeded operations
+Failed  | Number | Number of failed operations
+results | Object | Object containing result objects of each individual key
+&nbsp;
+
+```
+{
+    success: "Number of succeeded operations",
+    failed: "Number of failed operations",
+    results: {
+        key1: {
+            result: "Result for key1",
+            error: "error for key1 if any"
+        },
+        key2: {
+            result: "Result for key2",
+            error: "error for key2 if any"
+        }, 
+        ...
+        keyN: {
+            result: "Result for keyN",
+            error: "error for keyN if any"
+        }
+    }
+}
+```
+
+### Redis Connection Object
+
+Sigma will return redis client connection object in callback. Connection object can be
+reused for further operations without creating new connection.
+
+**Important**
+> **Make sure to close connection after it is used. Otherwise your lambda may face connection time out**
```

Comment:
close _the_

after _its final use_.

_L_ambda

face _an execution timeout._
(I suppose that's what actually happens - not a connection timeout?)